Replace FailedToCommitClusterStateException with NotMasterException

joshua-adams-1 opened the pull request:
Replaces the `FailedToCommitClusterStateException` inside `MasterService .BatchingTaskQueue.submitTask` with a `NotMasterException`. Since this exception is thrown when the node is shutting down and closing its threadpools, it can no longer be the current master.


Relates to: ES-13061

joshua-adams-1 committed:
Replace FailedToCommitClusterStateException with NotMasterException

Replaces the FailedToCommitClusterStateException inside MasterService
.BatchingTaskQueue.submitTask with a NotMasterException.

Relates to: ES-13061

DaveCTurner reviewed (commented):
CI doesn't look very happy about this - please request my review again once that's addressed.

joshua-adams-1 committed:
Merge branch 'main' into batch-task-queue-failed-commit-exception

joshua-adams-1 committed:
Replace FailedToCommitClusterStateException with NotMasterException

Replaces the FailedToCommitClusterStateException inside MasterService
.BatchingTaskQueue.submitTask with a NotMasterException.

Relates to: ES-13061

joshua-adams-1 committed:
Merge branch 'batch-task-queue-failed-commit-exception' of https://github.com/joshua-adams-1/elasticsearch into batch-task-queue-failed-commit-exception

joshua-adams-1 committed:
Revert TransportClusterHealthAction change

DaveCTurner reviewed (approved):
LGTM (but please fix the top-level comment to be the commit message)
